src/System.Console/tests/SyncTextReader.cs

```diff
@@ -66,8 +66,15 @@ private static void Test(string content, Action action)
         }
         finally
         {
-            Console.SetOut(savedStandardOutput);
-            Console.SetIn(savedStandardInput);
+            using (TextWriter oldWriter = Console.Out)
```

Comment:
Can you add some comments around this using pattern here?  It's a little confusing to me since you don't use oldWriter in the body of the block (I think you are doing this for an exception safe dispose on the oldWriter?)